src/middlewares/validation.middleware.ts

```diff
@@ -60,6 +61,23 @@ async function compileAjv(options: ValidationMiddlewareOptions) {
   return ajvInstance.compile(schema);
 }
 
+async function validateRequestParameters(params: any, next:NextFunction) {
+  if (!params) return;
+  const valid = await isValidUUID(params.id);
+  if (valid === true) {
+    next();
+  }
+
+  if (valid === false) {
+    throw new ProblemException({
+      type: 'invalid-request-params',
+      title: 'Invalid Request Parameters',
+      status: 422,
+      validationErrors: 'Invalid Request Paramaters' as any,
+    });
+  }
+}
```

Comment:
Validation of parameters should be done in generic way, based on the openapi document, not hardcoded for `share/{id} ` path